kotlinpoet/build.gradle.kts

```diff
@@ -13,40 +13,59 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-tasks.jar {
-  manifest {
-    attributes("Automatic-Module-Name" to "com.squareup.kotlinpoet")
-  }
-}
-
-tasks.compileTestKotlin {
-  compilerOptions {
-    freeCompilerArgs.add("-opt-in=com.squareup.kotlinpoet.DelicateKotlinPoetApi")
-  }
+plugins {
+  kotlin("multiplatform")
 }
 
 spotless {
   kotlin {
     targetExclude(
       // Non-Square licensed files
-      "src/main/java/com/squareup/kotlinpoet/ClassName.kt",
-      "src/test/java/com/squareup/kotlinpoet/AbstractTypesTest.kt",
-      "src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt",
-      "src/test/java/com/squareup/kotlinpoet/TypesEclipseTest.kt",
-      "src/test/java/com/squareup/kotlinpoet/TypesTest.kt",
+      "src/commonMain/kotlin/com/squareup/kotlinpoet/ClassName.kt",
+      "src/commonTest/kotlin/com/squareup/kotlinpoet/AbstractTypesTest.kt",
+      "src/commonTest/kotlin/com/squareup/kotlinpoet/ClassNameTest.kt",
+      "src/commonTest/kotlin/com/squareup/kotlinpoet/TypesEclipseTest.kt",
+      "src/commonTest/kotlin/com/squareup/kotlinpoet/TypesTest.kt",
     )
   }
 }
 
-dependencies {
-  implementation(libs.kotlin.reflect)
-  testImplementation(libs.kotlin.junit)
-  testImplementation(libs.truth)
-  testImplementation(libs.guava)
-  testImplementation(libs.compileTesting)
-  testImplementation(libs.jimfs)
-  testImplementation(libs.ecj)
-  testImplementation(libs.kotlinCompileTesting)
-  testImplementation(libs.kotlin.annotationProcessingEmbeddable)
-  testImplementation(libs.kotlin.compilerEmbeddable)
+kotlin {
+  jvm {
+    withJava()
+  }
+
+  sourceSets {
+    val commonMain by getting {
+      dependencies {
+        implementation(libs.kotlin.reflect)
+      }
+    }
+    val commonTest by getting {
+      dependencies {
+        implementation(libs.kotlin.junit)
+        implementation(libs.truth)
+        implementation(libs.guava)
+        implementation(libs.compileTesting)
+        implementation(libs.jimfs)
+        implementation(libs.ecj)
+        implementation(libs.kotlinCompileTesting)
+        implementation(libs.kotlin.annotationProcessingEmbeddable)
+        implementation(libs.kotlin.compilerEmbeddable)
+      }
+    }
+  }
+}
+
+tasks.withType(org.gradle.jvm.tasks.Jar::class.java) {
+  manifest {
+    attributes("Automatic-Module-Name" to "com.squareup.kotlinpoet")
+  }
 }
+
+tasks.named<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>("compileTestKotlinJvm") {
+  compilerOptions {
+    freeCompilerArgs.add("-opt-in=com.squareup.kotlinpoet.DelicateKotlinPoetApi")
+  }
+}
```

Comment:
There's a DSL for this we can use in a follow-up

https://github.com/cashapp/molecule/blob/782985fcf92d6385d164bbed91184f99b1f9141e/molecule-runtime/build.gradle#L42-L46